service/src/test/java/io/nuvalence/workmanager/service/service/NoteServiceTest.java

```diff
@@ -309,4 +318,159 @@ void getFilteredNotes() {
+
+        Exception exception =
+                assertThrows(
+                        NotFoundException.class,
+                        () -> noteService.getNoteById(transactionId, noteId));
+
+        assertEquals(TRANSACTION_NOT_FOUND, exception.getMessage());
+    }
+
+    @Test
+    void testGetNoteByIdNoteNotFound() {
+        UUID transactionId = UUID.randomUUID();
+        Long noteId = 1L;
+
+        TransactionNote note = new TransactionNote();
+        List<UUID> documentList = new ArrayList<>();
+        note.setId(noteId);
+        note.setTransactionId(transactionId);
+        note.setTitle("title");
+        note.setBody("body");
+        note.setCreatedBy(UUID.randomUUID().toString());
+        note.setCreatedTimestamp(OffsetDateTime.now().minusDays(1));
+        note.setDocuments(documentList);
+        Transaction transaction = new Transaction();
+
+        when(transactionService.getTransactionById(transactionId))
+                .thenReturn(Optional.of(transaction));
+
+        when(noteRepository.findByTransactionIdAndId(transactionId, noteId))
+                .thenReturn(Optional.empty());
+
+        Exception exception =
+                assertThrows(
+                        NotFoundException.class,
+                        () -> noteService.getNoteById(transactionId, noteId));
+
+        assertEquals("Note not found", exception.getMessage());
+    }
+
+    @Test
+    void testCreateTransactionNoteTransactionNotFound() {
+        UUID transactionId = UUID.randomUUID();
+        Long noteId = 1L;
+        TransactionNote note = new TransactionNote();
+        note.setId(noteId);
+        when(noteService.getNoteTypeByName("testName")).thenReturn(Optional.of(new NoteType()));
+        when(transactionService.getTransactionById(transactionId)).thenReturn(Optional.empty());
+        Exception exception =
+                assertThrows(
+                        NotFoundException.class,
+                        () -> noteService.createTransactionNote(transactionId, note, "testName"));
+
+        assertEquals(TRANSACTION_NOT_FOUND, exception.getMessage());
+    }
+
+    @Test
+    void testCreateTransactionNoteWithNoDocuments() {
+        UUID transactionId = UUID.randomUUID();
+        Long noteId = 1L;
+
+        TransactionNote note = new TransactionNote();
+        note.setId(noteId);
+        note.setTransactionId(transactionId);
+        note.setTitle("title");
+        note.setBody("body");
+        note.setDocuments(null);
+        note.setCreatedBy(UUID.randomUUID().toString());
+        note.setCreatedTimestamp(OffsetDateTime.now().minusDays(1));
+        Transaction transaction = new Transaction();
+        when(noteService.getNoteTypeByName("testName")).thenReturn(Optional.of(new NoteType()));
+        when(transactionService.getTransactionById(transactionId))
+                .thenReturn(Optional.of(transaction));
+        when(noteRepository.save(any())).thenReturn(note);
```

Comment:
```suggestion
        when(noteRepository.save(any())).then(AdditionalAnswers.returnsFirstArg());
```
Adding this helps test what the method does, and in this case it possibly reveals an error. Recommend checking.

Reply:
suggestion applied